accepted/future-releases/nnbd/number-operation-typing.md

```diff
@@ -62,47 +62,76 @@ T add<T extends num>(T a, T b) => a + b;
+> For an expression `e` of the form  `e1 + e2`, `e1 - e2`, `e1 * e2`, `e1 % e2` or `e1.remainder(e2)`, if the static type of  `e1` is a non-`Never` subtype of `int`, if the context type of `e` is a non-`Nevers` subtype of `int`, then the context type of `e2` is `int`. If the static type of `e1` is a non-`Never` subtype of `num` that is *not* a subtype of `double` and the context type of`e` is `double`, then the context type of `e2` is `double`.
 
-> If the context type of `e1.clamp(e2, e3)` is a non-`Never` subtype of `num`, *T*, and the static type of `e1` is a non-`Never` subtype of *T*, then the context types of `e2` and `e3` are both *T*.<br>Otherwise, if the context type of the invocation is `int` and the static type of `e1` is a non-`Never` subtype of `int`, then the context types of `e2` and `e3` are both `int`, and if the context type of the invocation is `double` and the static type of `e1` is a non-`Never` subtype of `double`, then the context types of `e2` and `e3` are both `double`. Otherwise the context types of `e2` and `e3` are `num`.
+> If the context type of `e1.clamp(e2, e3)` is a non-`Never` subtype of `num`, *T*, then the context types of `e2` and `e3` are also *T*. Otherwise the context types of `e2` and `e3` are `num`.
 
 (This does emphasize the inherent non-symmetry of Dart operators: The first operand is a receiver which is always evaluated with no type context, and is then used to resolve the operator method against, and the second operand is an argument to that method. We need to fully resolve the first operand and the operator before we can even begin with the second operand.)
 
+Tabulated, the context type of the second operand, based on the first operand and the context type of the entire operation, is:
+
+| op1 \\ ctx  | int             | int+            | double | double+         | num  | num+            |
+| :---------: | --------------- | --------------- | ------ | --------------- | ---- | --------------- |
+|   **int**   | int             | int<sup>*</sup> | double | num<sup>*</sup> | num  | num<sup>*</sup> |
+|  **int+**   | int             | int             | double | num<sup>*</sup> | num  | num<sup>*</sup> |
+| **double**  | num<sup>*</sup> | num<sup>*</sup> | num    | num<sup>*</sup> | num  | num<sup>*</sup> |
+| **double+** | num<sup>*</sup> | num<sup>*</sup> | num    | num             | num  | num<sup>*</sup> |
+|   **num**   | num<sup>*</sup> | num<sup>*</sup> | double | num<sup>*</sup> | num  | num<sup>*</sup> |
+|  **num+**   | num<sup>*</sup> | num<sup>*</sup> | double | num<sup>*</sup> | num  | num             |
```

Comment:
Of course, all those `T+` types must denote the _same_ type when they occur as the type of the receiver as well as the context type. However, in the case where both the receiver static type and the context type is some `U <: num`, it would actually be sound to give `e` the static type `U`. So I guess that would be expressed as `num+` in the last column in line 131.

It would also be satisfied by the rules if we return to the two subtype based rules: `e1` has type `U`, the context type is `U`, and `e2` is subject to type inference with context type `U`, which succeeds and yields an `e2` of type `U`.